### ISSUE TITLE
realtimeSort log axis issue

### ISSUE THREAD

**Issue by 100pah** [apache/echarts#13678]
### Version
5.0.0

### Steps to reproduce
When the value is 0, NaN will be calculated in graphic element and attached text, and they will be positioned on the left-top most corner of the canvas. What's more, the the value become none-zero, the graphic elements can not render normally.

Example:
incubator-echarts-examples `next` branch:
http://127.0.0.1:3002/en/editor.html?c=covid-america
Need change the `yAxis.type` as `'log'`.

**Comment by 100pah** [comment]
![image](https://user-images.githubusercontent.com/1956569/104087075-17a30b80-5298-11eb-8654-1a7099814a90.png)


To fix this issue, we might:

+ See `LogScale['unionExtentFromData']`: we might add param to the method `List['getApproximateExtent']` and `List['getDataExtent']` to union data without that `<= 0`. Otherwise the logScale extent will not be correct (e.g., [-Infinity, 10]).
+ Either 
    + In `BarView`, check wether the el has invalid layout (has NaN) and mark as `ignore`.
    + In `zrender`, if the Path has invalid shape params (NaN), do not render them.
+ Fix the animation that from a NaN prop.

Test case:
```js
            require([
                'echarts'
            ], function (echarts) {
                var data = [0, 1, 10, 100, 1000];
                var option = {
                    xAxis: {
                        type: 'log'
                    },
                    yAxis: {
                        type: 'category',
                        data: ['a', 'b', 'c', 'd', 'e']
                    },
                    series: [{
                        type: 'bar',
                        data: data,
                        label: { show: true, position: 'right' }
                    }]
                };

                var chart = testHelper.create(echarts, 'main1', {
                    title: [
                        'click "all +10"',
                        'check the x extent should be correct',
                        'the 0 bar should be displayed with animation'
                    ],
                    option: option,
                    buttons: [{
                        text: 'all +10',
                        onclick: function () {
                            for (var i = 0; i < data.length; i++) {
                                data[i] += 10;
                            }
                            chart.setOption({
                                series: {
                                    data: data
                                }
                            });
                        }
                    }]
                });
            });

```

**Comment by Ovilia** [comment]
Fixed in #15208.
